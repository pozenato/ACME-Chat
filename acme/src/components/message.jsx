import React, { Component } from 'react';
import moment from 'moment';


export default class Message extends Component {

    constructor(props) {
        super()
    }

    handleSameMessage = (rightDirection, message) => (
        <div className="messageDisplay">
                <div className={rightDirection ? "right messageText" : "left messageText"}>{message.text}</div>
        </div> 
    )

    handleNewAuthorMessage = (rightDirection, message) => (
        <div className="messageDisplay">
                <div className={rightDirection ? "rightHeader messageHeader" : "leftHeader messageHeader"}>{<img className="messageImage" src={message.userMessage.image} alt="logo img"/>} {message.userMessage.name}, {moment(message.date).format('LT')}</div>
                <div className={rightDirection ? "right messageText" : "left messageText"}>{message.text}</div>
        </div>
    )

    render() {
        const {
            message,
            sameAuthor,
            sameHour,
            rightDirection,
        } = this.props;
        return (
            <div className="messageContext" id="inner">
                {
                    sameHour && sameAuthor ?
                        this.handleSameMessage(rightDirection, message)
                    :
                        this.handleNewAuthorMessage(rightDirection, message)
                }
            </div>

        )
    }
};