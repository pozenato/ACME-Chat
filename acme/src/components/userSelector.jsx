import React, { Component} from 'react';
import { Menu, Dropdown, Icon, Button, message } from 'antd';


class UserSelector extends Component {

  constructor(props) {
    super()
  }

  handleButtonClick = (e) => {
    message.info('Click on left button.');
  }

  handleMenuClick = (e) => {
    this.props.handleUserSelected(e.key)
  }

  menuItems = () => {
    if (this.props.users) {
      return (
        <Menu onClick={(this.handleMenuClick)} className="menuConteiner">
          {
            this.props.users.map(u => {
              return (
                <Menu.Item className="userItemMenu" key={u.id}>
                      <img className="userImage" src={u.image} alt="logo img"/>
                      <h4 className="nameItem">{u.name}</h4>
                </Menu.Item>
              )
            })
          }
        </Menu>
      )
    }
    return <Menu />;
  };

render() {

    const { userSelected, defaultText } = this.props;

    return (
      <div id="components-dropdown-userSelector-dropdown-button">
        <Dropdown className="DropDownMenu" overlay={this.menuItems()}>
          <Button>
            {
              userSelected.id > 0 ? 
              <div className="userItemMenu">
                <img className="userImage" src={userSelected.image} alt="logo img"/>
                  <h4 className="nameItem">{userSelected.name}</h4>
              </div> : 
              <div>
                {defaultText}
                <Icon className="searchIcon" type="down" />
              </div>
            }            
          </Button>
        </Dropdown>
      </div>
    );
  }
}

export default UserSelector;