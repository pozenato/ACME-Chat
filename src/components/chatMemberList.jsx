import React, { Component } from 'react'

class chatMemberList extends Component {

    constructor(props) {
        super()
    }

    fetchChatMembers = () => {
        let unique = null;
        let messageUsers = [];
        if (this.props.messages) {
            messageUsers = this.props.messages.map(m => m.userMessage);
            unique = [ ...new Set( messageUsers ) ];
        }
        return unique;
    }

    render() {
        const users = this.fetchChatMembers();

        return (
            <div>
                <h1 className="chatMemberTitle">Chat Members</h1>
                {
                    users !== null ?
                        users.map(m => (
                            <div className="chatMemberContext" key={m.id}>
                                <img className="chatMemberImage" src={m.image} alt="logo img"/>
                                <h4 className="chatMemberName">{m.name}</h4>
                            </div>
                        )) 
                    : 
                        null
                }
            </div>
        )
    }
}

export default chatMemberList;