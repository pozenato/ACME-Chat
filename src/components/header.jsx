import React from 'react';
import logo from '../../resources/images/Acme-corp.png';

const Header = () => (
    <div className="headerContainer">
        <div className="logoContent">
            <img className="headerImg" src={logo} alt="logo img"/>            
        </div>
        <div className="titleContent">
            <h1 className="headerText">ACME Discussions</h1>
        </div>
    </div>
)

export default Header;