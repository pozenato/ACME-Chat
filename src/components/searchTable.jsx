import React from 'react'
import { Table } from 'antd';

export default props => {

    const columns = [
        {
          title: 'Name',
          dataIndex: 'name',
          key: 'name',
        },
        {
          title: 'Date',
          dataIndex: 'date',
          key: 'date',
        },
        {
          title: 'Message',
          dataIndex: 'message',
          key: 'message',
        }
    ];
    
    return (
        <Table columns={columns} dataSource={props.messages} />
    );
}