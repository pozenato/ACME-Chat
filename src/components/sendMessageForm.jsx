import React, { Component } from 'react';
import component from '../components';
import { Input, Button } from 'antd';
import moment from 'moment';
import findUserSelected from '../utils/sharedFunctions';
import constants from '../utils/constants';


class SendMessageForm extends Component {

    constructor(props) {
        super(props)
        this.state = this.getInitialState()
    }

    getInitialState() {
        return {
            userSelected: 0,
            messageValue: '',
        }
    }

    handleUserSelected = user => {
        const userSelected = findUserSelected(user, this.props.users);
        this.setState({ userSelected });
    }

    handleSubmitMessage = () => {
        this.props.addNewMessage({            
            user: this.state.userSelected, 
            date: moment(),
            text: this.state.messageValue,
        }
        );
        this.setState({ messageValue: '' })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.handleSubmitMessage();
    }

    render() {
        const {
            userSelected,
            messageValue,
        } = this.state;
        const {
            users,
        } = this.props;
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="sendMessageContext">
                    <div className="selectorMessage">
                        <component.userSelector
                            userSelected={userSelected}
                            handleUserSelected={this.handleUserSelected}
                            users={users}
                            defaultText={constants.userSelector.sendMessageDefaultText}
                        />         
                    </div>
                    <Button shape="circle" type="primary"  icon="search" onClick={this.props.showModal} />
                    <div className="inputMessage">
                        <Input
                            placeholder="Type a Message"
                            disabled={userSelected === 0}
                            value={messageValue}
                            onChange={e => this.setState({ messageValue: e.target.value })} />
                    </div>
                    <div className="sendButton">
                        <Button
                            type="primary"
                            shape="round"
                            icon="rocket"
                            size='large'
                            disabled={messageValue === ""}
                            onClick={this.handleSubmitMessage}
                        >
                            Send
                        </Button>
                    </div>
                </div>
            </form>
        );
    };

}

export default SendMessageForm;