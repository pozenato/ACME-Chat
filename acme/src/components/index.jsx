import chatMemberList from './chatMemberList';
import message from './message';
import messageList from './messageList';
import sendMessageForm from './sendMessageForm';
import userSelector from './userSelector';
import header from './header';
import searchTable from './searchTable';

export default {
    chatMemberList,
    message,
    messageList,
    sendMessageForm,
    userSelector,
    header,
    searchTable,
}