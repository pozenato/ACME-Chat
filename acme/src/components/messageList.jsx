import React, { Component } from 'react'
import components from '../components'

export default class MessageList extends Component {

    constructor(props) {
        super()
    }

    compareMinuteswithMessages = (currentDate, lastMessage) => {
        const minutes = (currentDate.diff(lastMessage.date, 'minutes'));
        return minutes < 1 ? true : false;
    }

    render() {
        const { messages } = this.props;
        let sameAuthor, sameHour = false;
        let rightDirection = true;
        let lastMessage = null;
        return (
            <div>
                {messages.map(m => {
                    if (lastMessage !== null) {
                        sameAuthor = m.userMessage.name === lastMessage.userMessage.name || false;
                        if (sameAuthor) {
                            sameHour = this.compareMinuteswithMessages(m.date, lastMessage);
                        } else {
                            sameHour = false;
                            rightDirection = !rightDirection;
                        }
                    }
                    lastMessage = m;
                    return (
                        <components.message
                            key={m.id}
                            message={m}
                            sameAuthor={sameAuthor}
                            sameHour={sameHour}
                            rightDirection={rightDirection}
                        />
                        )
                    }
                )}
            </div>
        )
    }
}