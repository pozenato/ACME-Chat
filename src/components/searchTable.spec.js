import React from 'react';
import components from '../components'
import renderer from 'react-test-renderer'

describe("The Search Table Component", ()=>{
    it("renders as expected", ()=> {
        const tree = renderer
        .create(<components.searchTable tags={[`css`, `html`, `go`]}/>)
        .toJSON();
        
        expect(tree).toMatchSnapshot();
    })
})