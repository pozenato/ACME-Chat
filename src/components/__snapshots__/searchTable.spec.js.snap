// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search Table Component renders as expected 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="ant-table-content"
        >
          <div
            className="ant-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": undefined,
                      "width": undefined,
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr
                  style={
                    Object {
                      "height": null,
                    }
                  }
                >
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Name
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Date
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                  <th
                    className=""
                  >
                    <span
                      className="ant-table-header-column"
                    >
                      <div>
                        <span
                          className="ant-table-column-title"
                        >
                          Message
                        </span>
                        <span
                          className="ant-table-column-sorter"
                        />
                      </div>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              />
            </table>
          </div>
          <div
            className="ant-table-placeholder"
          >
            <div
              className="ant-empty ant-empty-normal"
            >
              <div
                className="ant-empty-image"
              >
                <img
                  alt="No Data"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHZpZXdCb3g9IjAgMCA2NCA0MSIgIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                />
              </div>
              <p
                className="ant-empty-description"
              >
                No Data
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
